ra-optimisation/linear-range-action-rao/src/main/java/com/farao_community/farao/linear_range_action_rao/LinearRangeActionRao.java

```diff
@@ -34,8 +47,77 @@ public String getVersion() {
     }
 
     @Override
-    public CompletableFuture<RaoComputationResult> run(Network network, Crac crac, String variantId, ComputationManager computationManager, RaoParameters parameters) {
-        // TODO : implement LinearRangeActionRao
-        return null;
+    public CompletableFuture<RaoComputationResult> run(Network network,
+                                                       Crac crac,
+                                                       String variantId,
+                                                       ComputationManager computationManager,
+                                                       RaoParameters parameters) {
+        crac.synchronize(network);
+        // sensi
+        SystematicSensitivityAnalysisResult analysisResult = SystematicSensitivityAnalysisService.runAnalysis(network, crac, computationManager);
+
+        LinearRangeActionRaoResult resultExtension = new LinearRangeActionRaoResult(LinearRangeActionRaoResult.SecurityStatus.SECURED);
+
+        // 1. do for pre
+        LOGGER.info("Building result for precontingency:");
+        Map<String, Double> preReferenceFlow = analysisResult.getPreReferenceMargin();
+        List<MonitoredBranchResult> monitoredBranchResults = buildMonitoredBranchResultListFromReferenceMargins(crac, preReferenceFlow, resultExtension, null);
+        PreContingencyResult preRao = new PreContingencyResult(monitoredBranchResults);
+
+        // 2. do for each contingency
+        Map<Contingency, Map<String, Double> > mapReferenceFlowsMap = analysisResult.getContingencyReferenceMarginsMap();
+        List<ContingencyResult> contingencyResultsRao = new ArrayList<>();
+        for (Map.Entry<Contingency, Map<String, Double> > entry : mapReferenceFlowsMap.entrySet()) {
+            Contingency contingency = entry.getKey();
+            Map<String, Double> currentReferenceFlowMap = mapReferenceFlowsMap.get(contingency);
+
+            String idContSensi = contingency.getId();
+            String nameContSensi = contingency.getName();
+            LOGGER.info("Building result for post-contingency {}:", idContSensi);
+
+            List<MonitoredBranchResult> tmpMonitoredBranchResults = buildMonitoredBranchResultListFromReferenceMargins(crac, currentReferenceFlowMap, resultExtension, contingency);
```

Comment:
names are confusing. The variable which is here `currentReferenceFlowMap` is called `referenceMarginsMap` in the function `buildMonitoredBranchResultListFromReferenceMargins()`.

Reply:
--> i think it is a `referenceMarginsMap`in both cases

Reply:
--> but maybe I misunderstood

Reply:
ok, renamed